kingpin/actors/rightscale/server_array.py

```diff
@@ -549,24 +613,27 @@ def _execute(self):
             else:
                 self.log.info('Would enable array "%s"' % array.soul['name'])
 
-        # Launch all of the instances we want as quickly as we can. Note, we
-        # don't actually store the result here because we don't care about the
-        # returned instances themselves. If we launch 10, and 1 fails, we will
-        # rely on RightScale to re-launch that 1 host, rather than handing it
-        # in-code. Instead, our 'launch clicking' here is just a way to get the
-        # ball rolling as quickly as possible before rightscales
-        # auto-array-scaling kicks in.
-        self.log.info(
-            'Launching Array "%s" instances' % self.option('array'))
+    @gen.coroutine
+    def _execute(self):
+        # First things first, login to RightScale asynchronously to
+        # pre-populate the API attributes that are dynamically generated. This
+        # is a hack, and in the future should likely turn into a smart
+        # decorator.
+        yield self._client.login()
```

Comment:
Could have the base class call "self._preexecute()" inside of the real `self.execute()`